main.go

```diff
@@ -211,21 +230,78 @@ func main() {
 	log.Infoln("Received SIGTERM, exiting gracefully...")
 }
 
-func extURL(s string) (*url.URL, error) {
-	if s == "" {
+type peerDescSlice []mesh.PeerDescription
+
+func (s peerDescSlice) Len() int           { return len(s) }
+func (s peerDescSlice) Less(i, j int) bool { return s[i].UID < s[j].UID }
+func (s peerDescSlice) Swap(i, j int)      { s[i], s[j] = s[j], s[i] }
+
+// meshWait returns a function that inspects the current peer state and returns
+// a duration of one base timeout for each  peer with a higher ID than ourselves.
+func meshWait(r *mesh.Router, timeout time.Duration) func() time.Duration {
+	return func() time.Duration {
+		var peers peerDescSlice
+		for _, desc := range r.Peers.Descriptions() {
+			peers = append(peers, desc)
+		}
+		sort.Sort(peers)
+
+		k := 0
+		for _, desc := range peers {
+			if desc.Self {
+				break
+			}
+			k++
+		}
+		log.Warnf("timeout multiplier: %d", k)
+
+		return time.Duration(k) * timeout
+	}
+}
+
+func initMesh(addr, hwaddr, nickname string) *mesh.Router {
+	host, portStr, err := net.SplitHostPort(addr)
+
+	if err != nil {
+		log.Fatalf("mesh address: %s: %v", addr, err)
+	}
+	port, err := strconv.Atoi(portStr)
+	if err != nil {
+		log.Fatalf("mesh address: %s: %v", addr, err)
+	}
+
+	name, err := mesh.PeerNameFromString(hwaddr)
+	if err != nil {
+		log.Fatalf("%s: %v", hwaddr, err)
```

Comment:
It could be clearer what's wrong here, I'd not expect everyone to recognise a MAC address